web/src/stream_popover.ts

```diff
@@ -446,8 +447,9 @@ export async function build_move_topic_to_stream_popover(
         // topic and thus submit button is disabled if stream is also
         // not changed.
         util.the($<HTMLButtonElement>("#move_topic_modal button.dialog_submit_button")).disabled =
-            Number.parseInt(current_stream_id, 10) === select_stream_id &&
-            (new_topic_name === undefined || new_topic_name.trim() === old_topic_name.trim());
+            (realm.realm_mandatory_topics && !new_topic_name?.trim()) ||
+            (Number.parseInt(current_stream_id, 10) === select_stream_id &&
+                (new_topic_name === undefined || new_topic_name.trim() === old_topic_name.trim()));
```

Comment:
Might be more readable to have the empty topic condition in a separate if block.

Reply:
okh sure.